server/src/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -476,6 +476,25 @@ public VolumeVO doInTransaction(TransactionStatus status) {
         });
     }
 
+    /**
+     * Retrieves the volume name from CreateVolumeCmd object.
+     *
+     * If the retrieved volume name is null, empty or blank, then A random name
+     * will be generated using getRandomVolumeName method.
+     *
+     * @param cmd
+     * @return Either the retrieved name or a random name.
+     */
+    public String getVolumeNameFromCommand(CreateVolumeCmd cmd) {
+        String userSpecifiedName = cmd.getVolumeName();
+
+        if (org.apache.commons.lang.StringUtils.isBlank(userSpecifiedName)) {
```

Comment:
The code LGTM (I did not test it...just reviewed the code). My only comment is on this line. Curious why we don't provide the import at the top as is standard in CloudStack. Not a big deal, but it makes it easier to see at a glance what all packages are involved. Perhaps there is another StringUtils being imported, so this had to be done here?